File: rx-codelab/src/main/java/com/jraska/rx/codelab/http/UserCache.java
package com.jraska.rx.codelab.http;

import io.reactivex.Observable;

/**
 * Showcase purpose - not a cache at all :P
 */
public class UserCache {
  public static Observable<User> getUser(String login) {
    return Observable.fromCallable(() -> getUserSync(login));
  }

  private static User getUserSync(String login) {
    return new User(login, "", true, "");
  }
}

File: rx-codelab/src/main/java/com/jraska/rx/codelab/http/UserCache.kt
package com.jraska.rx.codelab.http

import io.reactivex.Observable

/**
 * Showcase purpose - not a cache at all :P
 */
object UserCache {
  fun getUser(login: String): Observable<User> {
    return Observable.fromCallable { getUserSync(login) }
  }

  private fun getUserSync(login: String): User {
    return User(login, "", true, "")
  }
}
